src/Uno.UI.RuntimeTests/Tests/Windows_Storage/Given_FileIO_Native.base.cs

```diff
@@ -0,0 +1,353 @@
+		{
+			var rootFolder = await GetRootFolderAsync();
+			StorageFile? targetFile = null;
+			try
+			{
+				var originalContent = "First line3527535205šěššýétžščžíé";
+				var appendedContent = "New line33251164535205věšřěšřčžčšžž";
+
+				targetFile = await rootFolder.CreateFileAsync(GetRandomTextFileName(), CreationCollisionOption.ReplaceExisting);
+				await FileIO.WriteTextAsync(targetFile, originalContent, Windows.Storage.Streams.UnicodeEncoding.Utf16BE);
+				await FileIO.AppendTextAsync(targetFile, appendedContent, Windows.Storage.Streams.UnicodeEncoding.Utf16BE);
+
+				var realContents = await FileIO.ReadTextAsync(targetFile, Windows.Storage.Streams.UnicodeEncoding.Utf16BE);
+				Assert.AreEqual(originalContent + appendedContent, realContents);
+			}
+			finally
+			{
+				await DeleteIfNotNullAsync(targetFile);
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+		public async Task When_AppendLinesAsyncNoEncoding()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			StorageFile? targetFile = null;
+			try
+			{
+				var firstLines = new[]
+				{
+					"First line",
+					"3527535205",
+					"šěššýétžščžíé"
+				};
+				var appendedLines = new[]
+				{
+					"New line",
+					"33251164535205",
+					"věšřěšřčžčšžž"
+				};
+
+				targetFile = await rootFolder.CreateFileAsync(GetRandomTextFileName(), CreationCollisionOption.ReplaceExisting);
+				await FileIO.WriteLinesAsync(targetFile, firstLines);
+				await FileIO.AppendLinesAsync(targetFile, appendedLines);
+
+				var realContents = await FileIO.ReadLinesAsync(targetFile);
+				CollectionAssert.AreEqual(firstLines.Concat(appendedLines).ToArray(), realContents.ToArray());
+			}
+			finally
+			{
+				await DeleteIfNotNullAsync(targetFile);
+				await CleanupRootFolderAsync();
+			}
+		}
+
+		[TestMethod]
+		public async Task When_AppendLinesAsyncRecognizesEncoding()
+		{
+			var rootFolder = await GetRootFolderAsync();
+			StorageFile? targetFile = null;
+			try
+			{
+				var firstLines = new[]
+				{
+					"First line",
+					"3527535205",
+					"šěššýétžščžíé"
+				};
+				var appendedLines = new[]
+				{
+					"New line",
+					"33251164535205",
+					"věšřěšřčžčšžž"
+				};
+
+				targetFile = await rootFolder.CreateFileAsync(GetRandomTextFileName(), CreationCollisionOption.ReplaceExisting);
+				await FileIO.WriteLinesAsync(targetFile, firstLines, Windows.Storage.Streams.UnicodeEncoding.Utf16BE);				
+				await FileIO.AppendLinesAsync(targetFile, appendedLines);
+
+				var realContents = await FileIO.ReadLinesAsync(targetFile, Windows.Storage.Streams.UnicodeEncoding.Utf16BE);
+
+				this.Log().LogWarning("Expected: " + string.Join("|", firstLines.Concat(appendedLines)));
+				this.Log().LogWarning("Actual: " + string.Join("|", realContents));
+				CollectionAssert.AreEqual(firstLines.Concat(appendedLines).ToArray(), realContents.ToArray());
+			}
+			finally
+			{
+				//await DeleteIfNotNullAsync(targetFile);
```

Comment:
This line and the `LogWarning` from above seem like debugging artifacts. To remove?